tweepy/api.py

```diff
@@ -219,7 +219,14 @@ def media_upload(self, filename, *args, **kwargs):
             :allowed_param:
         """
         f = kwargs.pop('file', None)
-        headers, post_data = API._pack_image(filename, 4883,
+
+        file_type = imghdr.what(filename)
+        if file_type == 'gif':
+            max_size = 14648
```

Comment:
This should actually be 14649. @MichaelCurrin actually rounded 14648.4375 down to 14648 rather than rounding up to 14649 as it says. This would cause GIFs between 14,999,553 (14648 * 1024 + 1) and 15,000,000 bytes (inclusive) to error even though Twitter's API should accept them.
```suggestion
            max_size = 14649
```

Reply:
Whoops! Always check the math :wink: 

Fixed.